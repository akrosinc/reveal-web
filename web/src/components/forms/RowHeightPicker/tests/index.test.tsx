import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import { Props, RowHeightFilter } from '..';

describe('src/components/forms/FilterForm/rowHeightFilter', () => {
  it('renders without crashing', () => {
    mount(<RowHeightFilter />);
  });
  it('works correctly', () => {
    // click on a label and check that changeHandler was called.
    const onChangeHandlerMock = jest.fn();
    const props: Props = {
      changeHandler: onChangeHandlerMock,
      rowHeightsConfs: {
        SHORT: {
          label: 'Short',
          value: '1em',
        },
        TALL: {
          label: 'Tall',
          value: '2em',
        },
      },
    };
    const wrapper = mount(<RowHeightFilter {...props} />);
    expect(wrapper.text()).toMatchSnapshot('rendered nodes');

    // lets click on one radio say the one with label short
    const nodeShort = wrapper.find('input#Short');
    nodeShort.simulate('change');
    wrapper.update();
    expect(onChangeHandlerMock).toHaveBeenCalledWith(props.rowHeightsConfs.SHORT.value);
    expect(toJson(wrapper.find('input[checked]'))).toMatchSnapshot(
      '#Short radio button is checked'
    );
  });
});
