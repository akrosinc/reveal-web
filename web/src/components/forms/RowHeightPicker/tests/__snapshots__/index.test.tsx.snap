// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/forms/FilterForm/rowHeightFilter works correctly: #Short radio button is checked 1`] = `
Array [
  <input
    checked={true}
    className="form-check-input"
    id="Short"
    name="rowHeight"
    onChange={[Function]}
    type="radio"
    value="1em"
  />,
  <input
    checked={false}
    className="form-check-input"
    id="Tall"
    name="rowHeight"
    onChange={[Function]}
    type="radio"
    value="2em"
  />,
]
`;

exports[`src/components/forms/FilterForm/rowHeightFilter works correctly: rendered nodes 1`] = `"Row Height Row HeightShortTall"`;
